### PR TITLE
Add setImmediate and setTimeout to Hermes example

### PR DESCRIPTION
**Pull request by vmoroz**
Extended `HermesRuntime` class to support limited `setImmediate`, `setTimeout`, and `console.log` polyfills.
Changed the example code in `Program` to rely on the `await` keyword.

### CODE REVIEW

**Review by jasongin — APPROVED** [review]

File: examples/hermes-engine/HermesRuntime.cs [hunk 1]
```diff
@@ -1,40 +1,300 @@
...
-            hermes_create_runtime((hermes_config)config, out _runtime).ThrowIfFailed();
-        }
-        else
+            taskPromise.TrySetResult(new HermesRuntime(dispatcherQueue));
+        });
+        return taskPromise.Task;
+    }
+
+    public Task RunAsync(Action action)
+    {
```
[the remark this answers is not in the document]
Comment (vmoroz) [reply examples/hermes-engine/HermesRuntime.cs:47]: Added `IsDisposed` check. I am going to get back to the code to see if we need more checks added.

File: examples/hermes-engine/HermesRuntime.cs [hunk 2]
```diff
@@ -1,40 +1,300 @@
...
     public static explicit operator hermes_runtime(HermesRuntime value) => value._runtime;
 
     public static explicit operator napi_env(HermesRuntime value)
-        => hermes_get_node_api_env((hermes_runtime)value, out napi_env env).ThrowIfFailed(env);
+    {
+        VerifyElseThrow(JSDispatcherQueue.GetForCurrentThread() == value._dispatcherQueue);
+        return hermes_get_node_api_env((hermes_runtime)value, out napi_env env).ThrowIfFailed(env);
+    }
+
+    public void CreatePolyfills()
```
[the remark this answers is not in the document]
Comment (vmoroz) [reply examples/hermes-engine/HermesRuntime.cs]: Changed to private.

File: examples/hermes-engine/HermesRuntime.cs [hunk 3]
```diff
@@ -1,40 +1,300 @@
...
+        using var scope = new JSValueScope();
+
+        // Add global
+        JSValue global = JSValue.Global;
+        global["global"] = global;
+
+        global["setImmediate"] = (JSCallback)(args =>
+        {
+            JSValue immediateCallback = args[0];
+            VerifyElseThrow(immediateCallback.TypeOf() == JSValueType.Function,
```
[the remark this answers is not in the document]
Comment (vmoroz) [reply examples/hermes-engine/HermesRuntime.cs]: Good point! Fixed.

[the remark this answers is not in the document]
Comment (vmoroz) [reply examples/hermes-engine/HermesRuntime.cs]: Cleaned up the code a bit to use new functions `AsFunction` and `AsInt32`. They throw `TypeError`.

File: examples/hermes-engine/HermesRuntime.cs [hunk 4]
```diff
@@ -100,45 +105,65 @@ public void CreatePolyfills()
...
+            int taskId = AsInt32(args, 0);
+            RemoveTimerTask(taskId);
             return default;
         });
 
         var console = new JSObject();
         console["log"] = (JSCallback)(args =>
         {
-            Console.WriteLine((string)args[0]);
+            Console.WriteLine((string)args[0].CoerceToString());
```
Comment (jasongin) [inline examples/hermes-engine/HermesRuntime.cs:138]: `JSValue.ToString()` already calls the JS `toString()`, so I think the cast / coercion is not necessary.

**Review by jasongin — APPROVED** [review]

**Pull request merged**